swagger.yaml

```diff
@@ -1069,6 +1176,60 @@ definitions:
             description: "The unique identifier for a collection of datasets that this dataset belongs"
             type: string
       - $ref: "#/definitions/Version"
+  NewInstance:
+    description: "A model for the request and response body for creating a new instance"
+    type: object
+    properties:
+      id:
+        type: string
+        readOnly: true
+        description: "A unique id for an instance"
+      dimensions:
+        type: array
+        description: "A list of codelists for each dimension of this instance"
+        items:
+          $ref: '#/definitions/Codelist'
+      links:
+        type: object
+        properties:
+          dataset:
+            description: "An object describing the ID and URL for a dataset which is associated with this instance"
+            type: object
+            properties:
+              id:
+                type: string
+                description: "The ID of the dataset associated with this instance"
+                example: "95c4669b-3ae9-4ba7-b690-87e890a1c67c"
+              href:
+                type: string
+                description: "The URL for the dataset associated with this instance"
+                example: "http://localhost:22000/datasets/95c4669b-3ae9-4ba7-b690-87e890a1c67c"
+          job:
+            description: "An object describing the ID and URL of the job containing this instance"
+            type: object
+            required: ["id", "href"]
+            properties:
+              id:
+                type: string
+                description: "The ID of the job containing this instance"
+                example: 042e216a-7822-4fa0-a3d6-e3f5248ffc35
+              href:
+                type: string
+                description: "The URL for the job containing this instance"
+                example: "http://localhost:21800/jobs/042e216a-7822-4fa0-a3d6-e3f5248ffc35"
+          self:
+            description: "An object describing the ID and URL for a dataset which is associated with this instance"
+            type: object
+            readOnly: true
+            properties:
+              href:
+                type: string
+                description: "The URL for this resource"
+                example: "http://localhost:22000/instances/45c4669b-3ae9-4ba7-b690-87e890a1c67f"
+      state:
+        description: "The state of the resource, this can only have a value of `created`"
```

Comment:
Does this need a bit more context? As the import process modifies this state to be 'submitted' and 'completed', but a client can not modify it.

Reply:
This is the response of a POST request and therefore state will always be equal to `created`, even if it does get updated to some other state, the response still holds a `state` value of `created`. It should be clear to an authorised user(developer creating an application) that if they need to update the state of the resource they use a PUT request.